Indexing/SubstitutionTree.hpp

```diff
@@ -1364,10 +1385,19 @@ class SubstitutionTree final
 
       class RobUnification { 
         RobSubstitution _subs;
+        unsigned _queryBank;
```

Comment:
If you are trying to claw back some performance, you could 

```c++
template<unsigned queryBank, unsigned normInternalBank, unsigned internalBank>
class RobUnification {
};
```

propagate the template "up", etc, etc, and then eventually return a `VirtualIterator` wrapping it as necessary.